lib/cqm/converter/bonnie_measure.rb

```diff
@@ -0,0 +1,179 @@
+          end
+          cql_library.statement_dependencies << statement_dependency
+        end
+
+        cqm_measure.cql_libraries << cql_library
+      end
+
+      cqm_measure.population_criteria = bonnie_measure.population_criteria
+      cqm_measure.data_criteria = bonnie_measure.data_criteria
+      cqm_measure.source_data_criteria = bonnie_measure.source_data_criteria
+      cqm_measure.measure_period = bonnie_measure.measure_period
+      cqm_measure.measure_attributes = bonnie_measure.measure_attributes
+
+      # convert populations, skipping strats to be added later
+      bonnie_measure.populations.each do |bonnie_population|
+        # skip if this is a stratification
+        next if bonnie_population.key?('stratification_index')
+
+        population_set = CQM::PopulationSet.new(
+          title: bonnie_population['title'],
+          id: bonnie_population['id']
+        )
+
+        # construct the population map and fill it
+        population_map = construct_population_map(cqm_measure)
+        bonnie_population.each_pair do |population_name, population_key|
+          # make sure it isnt metadata or an OBSERV or SDE list
+          next if ['id', 'title', 'OBSERV', 'supplemental_data_elements'].include?(population_name)
+
+          population_map[population_name.to_sym] = CQM::StatementReference.new(
+            library_name: cqm_measure.main_cql_library,
+            statement_name: get_cql_statement_for_bonnie_population_key(bonnie_measure.populations_cql_map, population_key)
+          )
+        end
+
+        population_set.populations = population_map
+
+        # add SDEs
+        if bonnie_population.key?('supplemental_data_elements')
+          bonnie_population['supplemental_data_elements'].each do |sde_statement|
+            population_set.supplemental_data_elements << CQM::StatementReference.new(
+              library_name: cqm_measure.main_cql_library,
+              statement_name: sde_statement
+            )
+          end
+        end
+
+        cqm_measure.population_sets << population_set
+      end
+
+      # add stratification info to population sets
+      bonnie_measure.populations.each do |bonnie_stratification|
+        next unless bonnie_stratification.key?('stratification_index')
+
+        cqm_measure.population_sets[bonnie_stratification['population_index']].stratifications << CQM::Stratification.new(
+          title: bonnie_stratification['title'],
+          id: bonnie_stratification['id'],
+          statement: CQM::StatementReference.new(
+            library_name: cqm_measure.main_cql_library,
+            statement_name: get_cql_statement_for_bonnie_population_key(bonnie_measure.populations_cql_map, bonnie_stratification['STRAT'])
+          )
+        )
+      end
+
+      # add observation info
+      bonnie_measure.observations.each do |bonnie_observation|
```

Comment:
I think we need to first check if `observations` exists.

Reply:
Added safe navigation operators.